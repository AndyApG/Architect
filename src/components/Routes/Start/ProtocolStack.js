import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Tweened from '../../../behaviours/Tweened';
import protocolCover from '../../../images/protocol-cover.png';

const Stack = Tweened(() => (
  <div className="start-protocol-stack__stack">
    <div className="start-protocol-stack__stack-cover">
      <img src={protocolCover} alt="" />
    </div>
  </div>
));

const getPath = ({ advanced, workingPath, archivePath }) =>
  encodeURIComponent((advanced && workingPath) || archivePath);

const ProtocolStack = ({ protocol: { id, archivePath, workingPath, advanced } }) => (
  <Link
    component="div"
    className="start-protocol-stack"
    to={`/edit/${getPath({ advanced, workingPath, archivePath })}`}
  >
    <div className="start-protocol-stack__preview">
      <Stack
        tweenName="protocol"
        tweenElement={id}
      />
    </div>
    <div className="start-protocol-stack__label">
      <h3 className="start-protocol-stack__name">{ archivePath || workingPath }</h3>
    </div>
  </Link>
);

ProtocolStack.propTypes = {
  protocol: PropTypes.object.isRequired,
};

export default ProtocolStack;
