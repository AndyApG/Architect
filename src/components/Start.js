import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Button, Icon } from '../ui/components';
import { actionCreators as protocolsActions } from '../ducks/modules/protocols';

const Welcome = ({
  show,
  createAndLoadProtocol,
  openProtocol,
}) => (
  <div className={cx('start', { 'start--hide': !show })}>
    <div className="start__welcome">
      <h1 className="start__welcome-title">Architect</h1>
      <h2 className="start__welcome-lead">A tool for creating Network Canvas interviews</h2>
    </div>

    <div className="start__call-to-action">
      <Button
        id="create-new-protocol-button"
        type="button"
        color="platinum"
        size="small"
        icon={<Icon name="arrow-right" color="charcoal" />}
        onClick={createAndLoadProtocol}
      >Create new</Button>
      <Button
        id="open-existing-protocol-button"
        type="button"
        size="small"
        color="cyber-grape"
        icon={<Icon name="arrow-right" />}
        onClick={openProtocol}
      >Open existing</Button>
    </div>
  </div>
);

Welcome.propTypes = {
  show: PropTypes.bool,
  createAndLoadProtocol: PropTypes.func.isRequired,
  openProtocol: PropTypes.func.isRequired,
};

Welcome.defaultProps = {
  show: true,
};

const mapDispatchToProps = dispatch => ({
  createAndLoadProtocol: bindActionCreators(protocolsActions.createAndLoadProtocol, dispatch),
  openProtocol: bindActionCreators(protocolsActions.openProtocol, dispatch),
});

export { Welcome };

export default connect(null, mapDispatchToProps)(Welcome);
