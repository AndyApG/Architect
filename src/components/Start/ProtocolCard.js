import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import protocolCover from '../../images/protocol-cover.png';

const ProtocolCard = ({ path, name }) => (
  <Link
    component="div"
    className="start-protocol-card"
    to={`/edit/${encodeURIComponent(path)}`}
  >
    <div className="start-protocol-card__preview">
      <div className="start-protocol-card__preview-cover">
        <img src={protocolCover} alt="" />
      </div>
    </div>
    <h3 className="start-protocol-card__name">{name}</h3>
    <div className="start-protocol-card__path">{path}</div>
  </Link>
);

ProtocolCard.propTypes = {
  path: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default ProtocolCard;
