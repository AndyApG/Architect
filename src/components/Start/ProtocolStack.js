import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Tweened from '../../behaviours/Tweened';
import protocolCover from '../../images/protocol-cover.png';

const Stack = Tweened(() => (
  <div className="start-protocol-card__stack">
    <div className="start-protocol-card__stack-cover">
      <img src={protocolCover} alt="" />
    </div>
  </div>
));

const getPath = ({ advanced, workingPath, archivePath }) =>
  encodeURIComponent((advanced && workingPath) || archivePath);

const ProtocolStack = ({ protocol: { id, archivePath, workingPath, advanced } }) => (
  <Link
    component="div"
    className="start-protocol-card"
    to={`/edit/${getPath({ advanced, workingPath, archivePath })}`}
  >
    <div className="start-protocol-card__preview">
      <Stack
        tweenName="protocol"
        tweenElement={id}
      />
    </div>
    <div className="start-protocol-card__label">
      <h3 className="start-protocol-card__name">{ archivePath || workingPath }</h3>
    </div>
  </Link>
);

ProtocolStack.propTypes = {
  protocol: PropTypes.object.isRequired,
};

export default ProtocolStack;
