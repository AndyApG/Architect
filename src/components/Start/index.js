/* eslint-disable import/prefer-default-export */
export { default as ProtocolStack } from './ProtocolStack';
