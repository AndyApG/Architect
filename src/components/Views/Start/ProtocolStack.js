import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { get } from 'lodash';
import Tweened from '../../../behaviours/Tweened';
import protocolCover from '../../../images/protocol-cover.png';
import { actionCreators as protocolsActions } from '../../../ducks/modules/protocols';

const Stack = Tweened(() => (
  <div className="start-protocol-stack__stack">
    <div className="start-protocol-stack__stack-cover">
      <img src={protocolCover} alt="" />
    </div>
  </div>
));

const getFilename = (path = '') => get(path.match(/([^/\\]+)$/), 1, path);

const ProtocolStack = ({ importAndLoadProtocol, protocol: { filePath } }) => (
  <div
    component="div"
    className="start-protocol-stack"
    onClick={() => importAndLoadProtocol(filePath)}
  >
    <div className="start-protocol-stack__preview">
      <Stack
        tweenName="protocol"
        tweenElement={filePath}
      />
    </div>
    <div className="start-protocol-stack__label">
      <h3 className="start-protocol-stack__name">{ getFilename(filePath) }</h3>
    </div>
  </div>
);

ProtocolStack.propTypes = {
  protocol: PropTypes.object.isRequired,
  importAndLoadProtocol: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  importAndLoadProtocol: bindActionCreators(protocolsActions.importAndLoadProtocol, dispatch),
});

export { ProtocolStack };

export default connect(null, mapDispatchToProps)(ProtocolStack);
