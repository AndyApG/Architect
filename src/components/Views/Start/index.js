/* eslint-disable import/prefer-default-export */
export { default as ProtocolStack } from './ProtocolStack';
export { default } from './Start';
