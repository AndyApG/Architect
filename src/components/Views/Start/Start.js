import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { get } from 'lodash';
import { compose } from 'recompose';
import { Button, Icon } from '../../../ui/components';
import { actionCreators as protocolsActions } from '../../../ducks/modules/protocols';
import ProtocolStack from './ProtocolStack';
import networkCanvasBrand from '../../../images/network-canvas-brand.svg';

class Start extends PureComponent {
  static propTypes = {
    recentProtocols: PropTypes.array.isRequired,
    createAndLoadProtocol: PropTypes.func.isRequired,
    openProtocol: PropTypes.func.isRequired,
    clearDeadLinks: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired,
  };

  static defaultProps = {
    protocols: [],
  };

  componentDidMount() {
    this.props.clearDeadLinks();
  }

  openProtocol = (protocol) => {
    this.props.history.push(`/edit/${encodeURIComponent(protocol.id)}`);
  }

  render() {
    return (
      <div className="start">
        <div className="start__split">
          <div className="start__welcome">
            <h1 className="start__welcome-title">Architect</h1>
            <h2 className="start__welcome-lead">A tool for creating Network Canvas interviews</h2>
          </div>

          <div className="start__call-to-action">
            <Button
              id="create-new-protocol-button"
              type="button"
              color="platinum"
              size="small"
              icon={<Icon name="arrow-right" color="charcoal" />}
              onClick={this.props.createAndLoadProtocol}
            >Create new</Button>
            <Button
              id="open-existing-protocol-button"
              type="button"
              size="small"
              color="cyber-grape"
              icon={<Icon name="arrow-right" />}
              onClick={this.props.openProtocol}
            >Open existing</Button>
          </div>
        </div>

        <div className="start__split">
          { this.props.recentProtocols.length > 0 &&
            <div className="start__protocols">
              { this.props.recentProtocols.map((protocol, index) => (
                <div
                  className="start__protocols-protocol"
                  key={index}
                >
                  <ProtocolStack
                    protocol={protocol}
                  />
                </div>
              )) }
            </div>
          }
        </div>

        <div className="start__background" />
        <img className="start__brand" src={networkCanvasBrand} alt="" />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  recentProtocols: get(state, 'recentProtocols', []).slice(0, 3),
});

const mapDispatchToProps = dispatch => ({
  createAndLoadProtocol: bindActionCreators(protocolsActions.createAndLoadProtocol, dispatch),
  openProtocol: bindActionCreators(protocolsActions.openProtocol, dispatch),
  clearDeadLinks: () => {},
  // clearDeadLinks: bindActionCreators(protocolsActions.clearDeadLinks, dispatch),
});

export { Start };

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
)(Start);
