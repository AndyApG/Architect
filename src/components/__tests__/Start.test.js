/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { Start } from '../Start';

const mockProps = {
  createAndLoadProtocol: () => {},
  openProtocol: () => {},
};

describe('<Start />', () => {
  it('can render?', () => {
    const component = shallow(<Start {...mockProps} />);

    expect(component).toMatchSnapshot();
  });
});
