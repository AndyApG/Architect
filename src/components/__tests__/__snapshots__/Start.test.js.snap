// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Start /> can render? 1`] = `
<div
  className="start"
>
  <div
    className="start__welcome"
  >
    <div
      className="start__welcome-logo"
    >
      <img
        alt=""
        src="architect-logo-icon.svg"
      />
    </div>
    <div>
      <h1
        className="start__welcome-title"
      >
        Architect
      </h1>
      <h2
        className="start__welcome-lead"
      >
        A tool for creating Network Canvas interviews
      </h2>
      <Version />
    </div>
  </div>
  <div
    className="start__call-to-action"
  >
    <Button
      color="platinum"
      content=""
      disabled={false}
      icon={
        <Icon
          className=""
          color="charcoal"
          name="arrow-right"
          style={Object {}}
        />
      }
      iconPosition="left"
      id="create-new-protocol-button"
      onClick={[Function]}
      size=""
      type="button"
    >
      Create new
    </Button>
    <Button
      color="cyber-grape"
      content=""
      disabled={false}
      icon={
        <Icon
          className=""
          color=""
          name="arrow-right"
          style={Object {}}
        />
      }
      iconPosition="left"
      id="open-existing-protocol-button"
      onClick={[Function]}
      size=""
      type="button"
    >
      Open existing
    </Button>
  </div>
</div>
`;
